mh/basic.py

```diff
@@ -0,0 +1,374 @@
+#! /usr/bin/env python
+# -*- coding: utf-8 -*-
+"""Monster Hunter PAT Server module.
+
+    Monster Hunter 3 Server Project
+    Copyright (C) 2022  InusualZ
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import multiprocessing
+import sys
+import threading
+import socket
+import random
+import struct
+
+from time_utils import Timer
+
+try:
+    # Python 3
+    import selectors
+    import queue
+except ImportError:
+    # Python 2
+    import selectors2 as selectors
+    import Queue as queue
```

Comment:
Do you mind refactoring the Server class so that it has a "dumb" implementation using `select` and a `SelectorMixIn` _(a bit like the `ForkingMixIn` and `ThreadingMixIn` class)_? That way the code architecture doesn't depend on this dependency and it doesn't force regular _(python2)_ users to download an external dependency. You could then create the `SelectorMixIn` class only if this dependency is available.

Reply:
I don't know if you still want this in? I made the change that we agreed upon on discord, which was to include the `selectors2` dependency in the project since it's a single file with all the implementation (`externals/selectors2.py`).

Reply:
Could you please add another directory so that the path is `externals/selectors2/selectors2.py` so we can add the LICENSE file in that folder too. This structure will also allow us to use more complex dependencies using submodules if that's really needed.

Otherwise, I'm still OK for it to be included, as you said the MIT licence doesn't conflict with the project's AGPL licence.

Reply:
Done